Keas.Mvc/Services/IdentityService.cs

```diff
@@ -198,26 +198,17 @@ public async Task<User> GetByKerberos(string kerb)
                 }
             }
 
-            var ucdKerbPerson = ucdKerbResult.ResponseData.Results.First();
-
-            // find their email
-            var ucdContactResult = await clientws.Contacts.Get(ucdKerbPerson.IamId);
-
-            if (ucdContactResult.ResponseData.Results.Length == 0)
+            var ucdKerbPerson = ucdKerbResult.ResponseData.Results.First();
+
+            var email = await GetEmailForUser(ucdKerbPerson, clientws);
+
+            //There used to be code here where it retunred null if the contact result was zero. That was moved into the GetEmailForUser method above.
+            if (email == null)
             {
                 return null;
             }
 
-            var ucdContact = ucdContactResult.ResponseData.Results.First();
-            var rtValue = CreateUser(ucdContact.Email, ucdKerbPerson, ucdKerbPerson.IamId);
-
-            if (string.IsNullOrWhiteSpace(rtValue.Email))
-            {
-                if (!string.IsNullOrWhiteSpace(ucdKerbPerson.UserId))
-                {
-                    rtValue.Email = $"{ucdKerbPerson.UserId}@ucdavis.edu";
```

Comment:
@laholstege It was doing here.

Reply:
alright, well. i think it'd be better to have no email than one we're guessing on, but i don't feel strongly about it and if we were doing it before i guess it's fine